Chapter comments

BKSalman opened the pull request:
### Todos:
- [x] add chapter comments migrations
- [x] add chapter comments mapping migrations
- [x] get comments handler
- [x] create comment handler
- [x] delete comment handler
- [x] frontend
    - [x] get comments
    - [x] display comments
    - [x] a way to send a comment

BKSalman committed:
add chapter comments db migrations

add chapter comments db migrations

BKSalman committed:
add chapter comments routes and handlers

BKSalman committed:
generalize `Comment` component

BKSalman committed:
add comments to chapter page

add comments to Chapter page

mohad12211 commented on client/src/lib/components/Comment.svelte:
```diff
@@ -5,12 +5,17 @@
+    export let comment: ComicCommentResponse | ChapterCommentResponse;
     export let indent = 0;
-    let openChildren = false;
+    let openChildren = comment.parent_comment == null;
```
problem with this is that if you add a reply under a reply, it's collapsed by default

BKSalman replied:
I made it like this to not clutter the screen with nested replies, so only top level replies (not to be confused with comments) will be open.

also when sending a reply I will make it open the reply replied to

mohad12211 replied:
yeah good idea

mohad12211 commented on client/src/routes/[username]/[comic_slug]/[chapter_number]/+page.svelte:
```diff
@@ -1,30 +1,89 @@
+    async function sendComment(
+        e: SubmitEvent,
+        parent_comment_id: string | null,
+        comic_id: string
```
```suggestion
        chapter_id: string
```

mohad12211 commented on client/src/routes/[username]/[comic_slug]/[chapter_number]/+page.svelte:
```diff
@@ -1,30 +1,89 @@
+        }
+
+        const res = await fetch(
+            `http://localhost:6060/api/v1/comics/${comic_id}/comments`,
```
```suggestion
            `http://localhost:6060/api/v1/comics/chapters/${chapter_id}/comments`,
```

BKSalman replied:
wtf, how, it shouldn't have worked with this bug

mohad12211 replied:
yeah it didn't work when I tried it

BKSalman replied:
ah, yeah, I didn't check the correctness of this file, mb

mohad12211 reviewed (changes requested):
copy paste mistakes

mohad12211 commented on src/comics/chapters/chapter_comments/routes.rs:
```diff
@@ -0,0 +1,211 @@
+pub async fn delete_comment(
+    auth: AuthExtractor<{ UserRole::User as u32 }>,
+    State(state): State<Arc<InnerAppState>>,
+    Path(comment_id): Path<Uuid>,
```
this is picking the chapter id.
the route should be `/api/v1/comics/chapters/comics/:comment:id`

BKSalman replied:
done (it's `/api/v1/comics/chapters/comments/:comment_id` btw)

BKSalman committed:
UX improvements + small refactor

- open child replies when replying to a reply
- focus input when opening reply box
- replace forms with variable

BKSalman committed:
fix chapter_comments endpoints

mohad12211 reviewed (approved):
LGTM LGTM
